src/Sparrow/Json/JsonOperationContext.cs

```diff
@@ -1229,9 +1229,18 @@ public MemoryStream CheckoutMemoryStream()
             return _cachedMemoryStreams.Pop();
         }
 
+        private const long MemoryStreamCacheThreshold = Constants.Size.Megabyte;
+        private const int MemoryStreamCacheMaxCapacity = 512;
         public void ReturnMemoryStream(MemoryStream stream)
         {
+            //We don't want to hold big streams in the cache or have too big of a cache
+            if (stream.Capacity > MemoryStreamCacheThreshold || _cachedMemoryStreams.Count >= MemoryStreamCacheMaxCapacity)
```

Comment:
This still gives us a single context with 512MB of memory.
Let's limit this to 64 MB total?

Reply:
64MB total and without capacity limit or 64MB instead of 512 items and still remove large streams?